src/components/GestureComponents.tsx

```diff
@@ -1,63 +1,82 @@
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type ScrollView = typeof ScrollView & {
+  scrollTo(
+    y?: number | { x?: number; y?: number; animated?: boolean },
+    x?: number,
+    animated?: boolean
+  ): void;
+  scrollToEnd(options?: { animated: boolean }): void;
+};
+export const Switch = createNativeWrapper<RNSwitchProps>(RNSwitch, {
+  shouldCancelWhenOutside: false,
+  shouldActivateOnStart: true,
+  disallowInterruption: true,
+});
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type Switch = typeof Switch;
+export const TextInput = createNativeWrapper<RNTextInputProps>(RNTextInput);
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type TextInput = typeof TextInput;
+export const DrawerLayoutAndroid = createNativeWrapper<
+  PropsWithChildren<RNDrawerLayoutAndroidProps>
+>(RNDrawerLayoutAndroid, { disallowInterruption: true });
+// we use literal object since TS gives error when using RN's `positions`
+// @ts-ignore FIXME(TS) maybe this should be removed?
+DrawerLayoutAndroid.positions = { Left: 'left', Right: 'right' };
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type DrawerLayoutAndroid = typeof DrawerLayoutAndroid;
 
-module.exports = {
-  /* RN's components */
-  get ScrollView() {
-    return memoizeWrap(ReactNative.ScrollView, {
-      disallowInterruption: true,
-      shouldCancelWhenOutside: false,
-    });
-  },
-  get Switch() {
-    return memoizeWrap(ReactNative.Switch, {
-      shouldCancelWhenOutside: false,
-      shouldActivateOnStart: true,
-      disallowInterruption: true,
-    });
-  },
-  get TextInput() {
-    return memoizeWrap(ReactNative.TextInput);
-  },
-  get DrawerLayoutAndroid() {
-    const DrawerLayoutAndroid = memoizeWrap(ReactNative.DrawerLayoutAndroid, {
-      disallowInterruption: true,
-    });
-    DrawerLayoutAndroid.positions = ReactNative.DrawerLayoutAndroid.positions;
-    return DrawerLayoutAndroid;
-  },
-  get FlatList() {
-    if (!MEMOIZED.FlatList) {
-      const ScrollView = this.ScrollView;
-      MEMOIZED.FlatList = React.forwardRef((props, ref) => (
-        <ReactNative.FlatList
-          ref={ref}
-          {...props}
-          renderScrollComponent={(scrollProps) => (
-            <ScrollView {...scrollProps} />
-          )}
-        />
-      ));
-    }
-    return MEMOIZED.FlatList;
-  },
+export const FlatList = React.forwardRef<RNFlatList<any>, RNFlatListProps<any>>(
+  (props, ref) => (
+    <RNFlatList
+      ref={ref}
+      {...props}
+      renderScrollComponent={(scrollProps) => <ScrollView {...scrollProps} />}
+    />
+  )
+);
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type FlatList<ItemT> = React.ComponentType<
+  RNFlatListProps<ItemT> &
+    NativeViewGestureHandlerProperties &
+    React.RefAttributes<any>
```

Comment:
What does it do?

Reply:
This allows passing refs, they're excluded from props by default.